OrbitGl/TimeGraph.cpp

```diff
@@ -377,6 +383,88 @@ void TimeGraph::ProcessTimer(const TimerInfo& timer_info) {
   NeedsUpdate();
 }
 
+//-----------------------------------------------------------------------------
+void TimeGraph::ProcessOrbitFunctionTimer(const FunctionInfo* function,
+                                          const TimerInfo& timer_info) {
+  FunctionInfo::OrbitType type = function->type();
+  uint64_t time = timer_info.start();
+  CHECK(timer_info.registers_size() > 0);
+  uint64_t graph_id = timer_info.registers(0);
+
+  switch (type) {
+    case FunctionInfo::kOrbitTrackInt: {
+      int32_t value = static_cast<int32_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackInt64: {
+      int64_t value = static_cast<int64_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackUint: {
+      uint32_t value = static_cast<uint32_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackUint64: {
+      uint64_t value = static_cast<uint64_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackFloatAsInt: {
+      int32_t int_value = static_cast<int32_t>(timer_info.registers(1));
+      float value = *(reinterpret_cast<float*>(&int_value));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+      break;
```

Comment:
Maybe put this and the following `break`s out of the `}` like the previous ones, for consistency.